src/components/URLImportDialog/index.js

```diff
@@ -0,0 +1,255 @@
+            width: "100%", // Easier to hit on mobile.
+            marginTop: 10,
+            marginBottom: 0,
+        },
+    },
+    importError: {
+        background: theme.palette.error.main,
+        marginBottom: 25,
+
+        // Hover colors get weird after switching to error
+        // without this.
+        "&:hover": {
+            backgroundColor: theme.palette.error.dark,
+        },
+
+        [theme.breakpoints.down("sm")]: {
+            width: "100%", // Easier to hit on mobile.
+            marginTop: 10,
+            marginBottom: 0,
+        },
+    },
+}));
+
+const setupEvent = (event) => {
+    event.preventDefault();
+    event.stopPropagation();
+};
+
+const validURL = (possibleURL) => {
+    let retval = true;
+
+    try {
+        const u = new URL(possibleURL);
+
+        if (!["http:", "https:", "ftp:"].includes(u.protocol)) {
+            retval = false;
+        }
+    } catch (e) {
+        retval = false;
+    } finally {
+        return retval;
+    }
+};
+
+const URLImportTextField = (props) => {
+    const classes = useStyles();
+    const [uploadURL, setUploadURL] = useState("");
+    const [isValidURL, setIsValidURL] = useState(false);
+    const [hasFirstFocused, setHasFirstFocused] = useState(false);
+
+    const { addURLFn, intl } = props;
+
+    const errored = !isValidURL && hasFirstFocused;
+
+    const clickHandler = (event) => {
+        setupEvent(event);
+        addURLFn(event, uploadURL);
+        setUploadURL("");
+        setIsValidURL(false);
+    };
+
+    return (
+        <div className={classes.container}>
+            <TextField
+                fullWidth
+                id={buildID(ids.BASE_ID, ids.TEXTFIELD)}
+                placeholder={fmt(intl, "placeholder")}
+                helperText={
+                    isValidURL || (!isValidURL && !hasFirstFocused)
+                        ? fmt(intl, "helperText")
+                        : fmt(intl, "helperTextError")
+                }
+                InputLabelProps={{
+                    shrink: true,
+                }}
+                aria-label={fmt(intl, "textFieldAriaLabel")}
+                className={classes.urlField}
+                error={errored}
+                value={uploadURL}
+                onFocus={(e) => {
+                    setHasFirstFocused(true);
+                }}
+                onBlur={(e) => {
+                    // If it's empty, we don't care if it's already been focused on.
+                    if (uploadURL === "") {
+                        setHasFirstFocused(false);
+                    }
+                }}
+                onChange={(e) => {
+                    setupEvent(e);
+
+                    const possibleUploadURL = e.target.value.trim();
+
+                    if (!validURL(possibleUploadURL)) {
+                        setIsValidURL(false);
+                    } else {
+                        setIsValidURL(true);
+                    }
```

Comment:
Minor, but this could be simplified to something like:
```
const isValid = validURL(possibleUploadURL);
setIsValidURL(isValid);
```

Reply:
Changed it to `setIsValidURL(validURL(possibleUploadURL));`.